xrpl/models/ledger_objects/ledger_object.py

```diff
@@ -0,0 +1,192 @@
+        else:
+            if "ledger_entry_type" in value:
+                ledger_entry_type = value["ledger_entry_type"]
+                if (
+                    cls.get_ledger_object_type(ledger_entry_type).__name__
+                    != cls.__name__
+                    and cls.get_md_ledger_object_type(ledger_entry_type).__name__
+                    != cls.__name__
+                ):
+                    raise XRPLModelException(
+                        f"Using wrong constructor: using {cls.__name__} constructor "
+                        f"with ledger object type {value['ledger_entry_type']}."
+                    )
+                value = {**value}
+                del value["ledger_entry_type"]
+            return super(LedgerObject, cls).from_dict(value)
+
+    @classmethod
+    def get_ledger_object_type(
+        cls: Type[LedgerObject], ledger_object_type: str
+    ) -> Type[LedgerObject]:
+        """
+        Gets the correct ledger object type.
+
+        Args:
+            ledger_object_type: The ledger object type willing to get.
+
+        Returns:
+            Type[LedgerObject]: The correct ledger object type.
+
+        Raises:
+            XRPLModelException: If the Ledger Object type does not exist.
+        """
+        import xrpl.models.ledger_objects as ledger_object_models
+
+        ledger_object_types: Dict[str, Type[LedgerObject]] = {
+            lgr_obj.value: getattr(ledger_object_models, lgr_obj)
+            for lgr_obj in ledger_object_models.ledger_entry_type.LedgerEntryType
+        }
+        if ledger_object_type in ledger_object_types:
+            return ledger_object_types[ledger_object_type]
+
+        raise XRPLModelException(
+            f"{ledger_object_type} is not a valid Ledger Object type."
+        )
+
+    @classmethod
+    def get_md_ledger_object_type(
+        cls: Type[LedgerObject], ledger_object_type: str
+    ) -> Type[LedgerObject]:
+        """
+        Gets the ledger object type within fields prefixed with "MD and suffixed with
+        "Fields".
+
+        Args:
+            ledger_object_type: The object type willing to get.
+
+        Returns:
+            Type[LedgerObject]: The correct ledger object type.
+
+        Raises:
+            XRPLModelException: If the Ledger Object type does not exist.
+        """
+        import xrpl.models.ledger_objects as ledger_object_models
+
+        ledger_object_types: Dict[str, Type[LedgerObject]] = {
+            lgr_obj.value: getattr(ledger_object_models, f"MD{lgr_obj}Fields")
+            for lgr_obj in ledger_object_models.ledger_entry_type.LedgerEntryType
+        }
+        if ledger_object_type in ledger_object_types:
+            return ledger_object_types[ledger_object_type]
+
+        raise XRPLModelException(
+            f"{ledger_object_type} is not a valid Ledger Object type."
+        )
+
+    def __getitem__(self: LedgerObject, field_name: str) -> Any:
```

Comment:
Is there a reason why this function exists for this base class and not for transactions or requests? It seems unnecessary.